# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1392705225 -3600
#      Tue Feb 18 07:33:45 2014 +0100
# Branch gui-release
# Node ID 0f3bc7ccb8750d9158c61f6813dbff613c15eaf4
# Parent  3a509de8e7911c19ce47250741841b489e5b6953
update the completion list of the editor while typing

* file-editor-tab.cc (constructor): disable removing the list at start;
  (handle_cursor_moved): show new auto completion list when list is shown

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -134,17 +134,17 @@ file_editor_tab::file_editor_tab (const 
   _edit_area->setBraceMatching (QsciScintilla::StrictBraceMatch);
   _edit_area->setAutoIndent (true);
   _edit_area->setIndentationWidth (2);
   _edit_area->setIndentationsUseTabs (false);
 
   _edit_area->setUtf8 (true);
 
   // auto completion
-  _edit_area->setAutoCompletionSource (QsciScintilla::AcsAll);
+  _edit_area->SendScintilla (QsciScintillaBase::SCI_AUTOCSETCANCELATSTART, false);
 
   QVBoxLayout *edit_area_layout = new QVBoxLayout ();
   edit_area_layout->addWidget (_edit_area);
   edit_area_layout->addWidget (_status_bar);
   edit_area_layout->setMargin (0);
   setLayout (edit_area_layout);
 
   // connect modified signal
@@ -1595,13 +1595,16 @@ file_editor_tab::center_current_line ()
 
       _edit_area->SendScintilla (2613,first_line); // SCI_SETFIRSTVISIBLELINE
     }
 }
 
 void 
 file_editor_tab::handle_cursor_moved (int line, int col)
 {
+  if (_edit_area->SendScintilla (QsciScintillaBase::SCI_AUTOCACTIVE))
+    show_auto_completion (this);
+
   _row_indicator->setNum (line+1);
   _col_indicator->setNum (col+1);
 }
 
 #endif
